### ISSUE TITLE
`TypeError` when checking with 1.1.335

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#6383]
**Describe the bug**
Type checking https://github.com/bryanforbes/botus_receptus/blob/60e0291406f6f4468106179cfd21e525e3e77865/src/botus_receptus/db/utils.py gives the following error:

```
﻿﻿﻿﻿An internal error occurred while type checking file "/Users/bryan/Projects/botus_receptus/src/botus_receptus/db/utils.py": TypeError: Cannot read properties of undefined (reading 'start')
    at validateFunctionArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:11404:23)
    at validateCallForFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9249:32)
    at validateCallArgumentsForSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9083:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9001:44)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19792:20)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8998:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3841:43)
    at doForEachSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:726:9)
    at expandSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3829:13)
    at mapSubtypesExpandTypeVars (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3873:13)
    at validateCallArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8994:26)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12662:34)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19792:20)
    at handleSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12661:17)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12703:24)
    at mapSubtypes (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:433:32)
    at getTypeOfMagicMethodCall (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12697:28)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:2655:40)
    at mapSubtypes (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:433:32)
    at getTypeOfIterator (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:2636:30)
    at matchFunctionArgumentsToParameters (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:10047:25)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8721:38)
    at Array.forEach (<anonymous>)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8717:66)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19798:28)
    at validateOverloadedFunctionArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8715:9)
    at validateCallForOverloadedFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:28)
    at validateCallArgumentsForSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9095:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9001:44)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19798:28)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8998:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3841:43)
    at doForEachSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:726:9)
    at expandSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3829:13)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3870:17)
    at Array.forEach (<anonymous>)
    at mapSubtypesExpandTypeVars (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3869:22)
    at validateCallArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8994:26)
    at getTypeOfCall (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:7861:36)
    at getTypeOfExpression (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1027:30)
    at evaluateTypesForExpressionInContext (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:18771:21)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:853:13)
    at evaluateTypeForSubnode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19010:9)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:852:16)
    at a.getTypeResult (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at j.visitReturn (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:908:48)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:526:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at j.walkMultiple (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at j._walkStatementsAndReportUnreachable (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2695:18)
    at j.visitSuite (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:276:14)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at j.visitFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:643:18)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:442:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
Error performing analysis: TypeError: Cannot read properties of undefined (reading 'start')
    at validateFunctionArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:11404:23)
    at validateCallForFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9249:32)
    at validateCallArgumentsForSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9083:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9001:44)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19792:20)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8998:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3841:43)
    at doForEachSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:726:9)
    at expandSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3829:13)
    at mapSubtypesExpandTypeVars (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3873:13)
    at validateCallArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8994:26)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12662:34)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19792:20)
    at handleSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12661:17)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12703:24)
    at mapSubtypes (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:433:32)
    at getTypeOfMagicMethodCall (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:12697:28)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:2655:40)
    at mapSubtypes (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:433:32)
    at getTypeOfIterator (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:2636:30)
    at matchFunctionArgumentsToParameters (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:10047:25)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8721:38)
    at Array.forEach (<anonymous>)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8717:66)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19798:28)
    at validateOverloadedFunctionArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8715:9)
    at validateCallForOverloadedFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:28)
    at validateCallArgumentsForSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9095:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9001:44)
    at useSpeculativeMode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19798:28)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8998:24)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3841:43)
    at doForEachSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:726:9)
    at expandSubtype (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3829:13)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3870:17)
    at Array.forEach (<anonymous>)
    at mapSubtypesExpandTypeVars (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3869:22)
    at validateCallArguments (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8994:26)
    at getTypeOfCall (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:7861:36)
    at getTypeOfExpression (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1027:30)
    at evaluateTypesForExpressionInContext (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:18771:21)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:853:13)
    at evaluateTypeForSubnode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19010:9)
    at callback (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:852:16)
    at a.getTypeResult (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at j.visitReturn (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:908:48)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:526:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at forEach (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at j.walkMultiple (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at j._walkStatementsAndReportUnreachable (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2695:18)
    at j.visitSuite (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:276:14)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at j.walk (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:267:19)
    at j.visitFunction (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:643:18)
    at j.visit (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:442:29)
    at j.visitNode (/opt/homebrew/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool?
Command line

Which version?
1.1.335

**Comment by erictraut** [comment]
Thanks for the bug report. This was a long-standing bug that was exposed by a recent change I made. It will be addressed in the next release. You can work around it in the meantime by downgrading to 1.1.334. Apologies for any inconvenience.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
